source/api.rst

```diff
@@ -1,3 +1,50 @@
+
+.. note::
+    Along a similar theme, I think this section tries to do too much and, 
+    therefore doesn't really achieve anything well. The application programming 
+    interface `"defines interactions between multiple software intermediaries" 
+    <https://en.wikipedia.org/wiki/Application_programming_interface>`_, and 
+    your intermediaries here are the end-user and Simulink. Thus, this needs 
+    to be a user experience (i.e. Useful, Usable, Findable, Credible, Desirable 
+    Accessible and Valuable) and, at the same time, document the interface to 
+    Simulink, which is more focussed on the developer, rather than the user. It 
+    isn't super clear in here what is user focussed and additionally there's no 
+    easy way to trace which documentation belongs to the methods called in 
+    Simulink (because information about which classes the methods are called on 
+    is missing). It would be much easier to trace what code is being called in 
+    Simulink if they were just functions, in this case, which again calls in to 
+    question the choice of using classes in the architecture. 
+    
+    For this piece of documentation, I think it should focus on interactions 
+    with the user, therefore properties and methods that are consumed by 
+    Simulink or in the main function shouldn't be included. Also, improving the 
+    introductions to the classes by explaining their purpose better and 
+    integrating the constructor documentation, rather than it being documented 
+    as a separate method would be useful. Additionally, because these classes 
+    need configured, some pointers as to which parameters must be set, would be 
+    helpful, IMO. 
+    
```

Comment:
I agree, @akeeste this relates to regrouping public versus private methods and properties at the first cut. The second part requires re-evaluating the use of the classes altogether. This could result in more functions applied universally.